### ISSUE TITLE
[BUG] upgrader site has contradictory message, seemingly refuses to upgrade

### ISSUE THREAD

**Issue by twMat** [Jermolene/TiddlyWiki5#6869]
### Describe the bug

As reported [by a user](https://talk.tiddlywiki.org/t/upgrade-to-core-blocked/4158/8?u=twmat) it is confusing that the officially recommended upgrade mechanism on https://tiddlywiki.com/upgrade seemingly blocks to upgrade the very core plugin. (I explain why in the linked to post). This:

<img width="436" alt="image" src="https://user-images.githubusercontent.com/4534056/182416493-6d8bdb36-d29b-46da-b379-c73794020df6.png">

### To Reproduce

1. Drag an outdated TW onto the upgrader mechanism at https://tiddlywiki.com/upgrade

### TiddlyWiki Configuration

-

### Additional context

I note the upgrade wiki has a special Upgrade plugin so maybe, for that very url, there could be some exception to how this is handled?

**Comment by pmario** [comment]
hmm, The response

>Yes, that is what I’m seeing. That makes sense now. I initially was thinking that the upgrade would update my outdated modules, but it sounds like it just doesn’t import my outdated modules.

is right. ... The upgrader uses the same import mechanism as every other TW. That's good for maintainability. ... We shouldn't change the mechanism ... but ...

We should be able to replace https://github.com/Jermolene/TiddlyWiki5/blob/3fd2cfc33918e9f8f75bcdcb17dedeee8435c152/core/language/en-GB/Import.multids#L24
with a different text in the upgrade plugin. It already contains some adjustments for other core templates. 

@twMat What new text would you suggest?

**Comment by pmario** [comment]
May be: `Your plugin <<incoming>> will be upgraded with <<existing>>`

What do you think?

**Comment by twMat** [comment]
How about 

`Your plugin will be upgraded from <<incoming>> to <<existing>>`

:)

**Comment by Jermolene** [comment]
Changing the text of the status messages in the import listing will be quite an undertaking.

Perhaps we could add some text at the top of the listing that provides a better context for the messages rather than changing the messages themselves.

**Comment by pmario** [comment]
>Changing the text of the status messages in the import listing will be quite an undertaking.

I don't think so. .. We do have the upgrade plugin, which can overwrite the message. It's a translatable message. So it should be doable

**Comment by pmario** [comment]
It should be a simple fix, since the Upgrader is English Only at the moment. I don't think there is a translated version out there.

And if there is one, they can add the new tiddler to the edtiion and translate it. 

![image](https://user-images.githubusercontent.com/374655/182858472-c223fc4a-20f2-456d-9348-74b91967907a.png)

**Comment by twMat** [comment]
BTW, is this only relevant for the core plugin? What about snowwhite and vanilla?

**Comment by pmario** [comment]
>BTW, is this only relevant for the core plugin? What about snowwhite and vanilla?

They have a different text which imo does make sense

**Comment by pmario** [comment]
Or do you want to change all of them. There are here

https://github.com/Jermolene/TiddlyWiki5/blob/3fd2cfc33918e9f8f75bcdcb17dedeee8435c152/core/language/en-GB/Import.multids#L23-L34

**Comment by twMat** [comment]
OK, testing it again, I conclude that only the core plugin prompt is problematic.

<img width="641" alt="image" src="https://user-images.githubusercontent.com/4534056/182884950-482cb2c4-c230-42fe-8950-aa7e4e6d64b0.png">

**Comment by Jermolene** [comment]
It would be possible for the upgrader to overwrite the language strings in question, but that would make it impossible to translate the upgrader.

I really think we're over engineering the fix to this very minor problem. In the first instance we should be trying to fix this by improving the text in a general way that works both for imports and the upgrader.

**Comment by pmario** [comment]
>It would be possible for the upgrader to overwrite the language strings in question, but that would make it impossible to translate the upgrader.

There is a pending draft PR, that would allow us to translate plugins. ... So I don't see a real problem there in the future ;)

**Comment by pmario** [comment]
>I really think we're over engineering the fix to this very minor problem. In the first instance we should be trying to fix this by improving the text in a general way that works both for imports and the upgrader.

We will need to have a closer look. .. I think it may be possible to find a wording, that works for both cases in English. I'm not so sure if that will work for German.

**Comment by Jermolene** [comment]
> There is a pending draft PR, that would allow us to translate plugins. ... So I don't see a real problem there in the future ;)

I'd like to revisit how we handle translations generally once #6666 is merged. `<<lingo>>` should become a global function, rather than a text substitution-based macro as at present. I can't recall the details of the translatable plugin PR but it's likely to need revision.

**Comment by pmario** [comment]
>It would be possible for the upgrader to overwrite the language strings in question, but that would make it impossible to translate the upgrader.

If someone will create a translated upgrader they need to create a new edition. The translated language tiddler can be part of that edition. So it is possible to translate the upgrader. ... It will be a bit hacky though.

**Comment by pmario** [comment]
Possible text for "blocked plugins"

```
The existing plugin (v<<existing>>) is newer than the incoming plugin (v<<incoming>>)
```

I think this would make sense for the default import mechanism and the upgrader. @Jermolene what do you think?

**Comment by saqimtiaz** [comment]
> The existing plugin (v<<existing>>) is newer than the incoming plugin (v<<incoming>>)

The text also needs to be appropriate for when an upgrade is blocked due to the incoming version being the same as the existing version.

The message displayed used to be `Blocked plugin (due to incoming <<incoming>> being older than existing <<existing>>)` which users found misleading and confusing when the two version numbers were the same. It was therefore changed to the current message: `Blocked plugin (due to incoming <<incoming>> not being newer than existing <<existing>>).` 

(See #5059)

**Comment by pmario** [comment]
What about: 

```
Your plugin will be <<incoming>>
```

done

**Comment by twMat** [comment]
The upgrader page shows the latest version number at the *bottom* of the page. This can be moved or duplicated at the top:

<img width="940" alt="image" src="https://user-images.githubusercontent.com/4534056/183029332-cea95b1f-0a90-43df-a8c8-d2e11b0f22f7.png">

So @pmario 's idea with `Your plugin will be v<<incoming>>` in combo with this should make things very clear!

**Comment by twMat** [comment]
The core plugin *always* shows as the very first row in the table, right? Otherwise, it should.

**Comment by pmario** [comment]
>The core plugin always shows as the very first row in the table, right? Otherwise, it should.

I don't know. But if it would not, it would be really hard to change that. The importer UI and workflow is probably **the** most complex thing in TiddlyWiki.

**Comment by pmario** [comment]
>The upgrader page shows the latest version number at the bottom of the page. This can be moved or duplicated at the top, and with bigger font.

That should be a simple fix \o/

**Comment by pmario** [comment]
@Jermolene ... What do you think?

Start image: 

![image](https://user-images.githubusercontent.com/374655/183033895-5779b828-a0bd-488c-b923-93574df1fcf5.png)

Import image

![image](https://user-images.githubusercontent.com/374655/183034056-e3c6f593-3a2e-4015-ac6e-a7cc9f70508c.png)

**Comment by pmario** [comment]
>The upgrader page shows the latest version number at the bottom of the page. This can be moved or duplicated at the top, and with bigger font.

Moved this one to a separated PR: https://github.com/Jermolene/TiddlyWiki5/pull/6881

**Comment by twMat** [comment]
>Moved this one to a separated PR: https://github.com/Jermolene/TiddlyWiki5/pull/6881

I guess this means I can close this.

**Comment by pmario** [comment]
>I guess this means I can close this.

No. .. This will need to be a different PR, since we didn't reach consensus about the wording yet.

**Comment by twMat** [comment]
👍

**Comment by pmario** [comment]
I think we did update the text as discussed here.

<img width="1295" height="59" alt="Image" src="https://github.com/user-attachments/assets/0164e804-6052-4cb6-a3d0-ee0a3f005926" />

So close again.
